Foundation/Process.swift

```diff
@@ -241,7 +241,11 @@ open class Process: NSObject {
         if let env = environment {
             let nenv = env.count
             envp = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 1 + nenv)
-            envp.initialize(from: env.map { strdup("\($0)=\($1)") })
+            let evars = env.map { (arg) -> UnsafeMutablePointer<Int8>? in
+                let (key, value) = arg
+                return UnsafeMutablePointer(strdup("\(key)=\(value)"))
+            }
+            envp.initialize(from: evars, count: 1 + nenv)
```

Comment:
I was actually going to revert this change since there was some discussion on the closure variables that @parkera  raised (the let (key, value) = arg change).

Isnt the mapped size correct since its allocating 1 + env, but the actual issue is we need to ensure its set to NULL to terminate the **env array?